iterator.go

```diff
@@ -602,6 +613,26 @@ func (it *Iterator) parseItem() bool {
 		}
 	}
 
+	// Skip keys with banned prefixes.
+	for mi.Valid() {
```

Comment:
You should check this before the all versions check below.

Reply:
This is the correct place for skipping the banned prefixes because after skipping the banned keys, we still need to check if the keys are internal or the key's version is under is above readTs.